modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer.ts

```diff
@@ -102,18 +103,30 @@ export default class MultiIconLayer<DataT, ExtraPropsT = {}> extends IconLayer<
   draw(params) {
     const {sdf, smoothing, outlineWidth} = this.props;
     const {outlineColor} = this.state;
+    const outlineBuffer = outlineWidth ? DEFAULT_BUFFER * (1 - outlineWidth) : -1;
 
     params.uniforms = {
       ...params.uniforms,
       // Refer the following doc about gamma and buffer
       // https://blog.mapbox.com/drawing-text-with-signed-distance-fields-in-mapbox-gl-b0933af6f817
       buffer: DEFAULT_BUFFER,
-      outlineBuffer: outlineWidth ? Math.max(smoothing, DEFAULT_BUFFER * (1 - outlineWidth)) : -1,
+      outlineBuffer,
       gamma: smoothing,
       sdf: Boolean(sdf),
       outlineColor
     };
+
     super.draw(params);
+
+    // draw text without outline on top to ensure a thick outline won't occlude other characters
+    if (sdf && outlineWidth) {
+      const {iconManager} = this.state;
+      const iconsTexture = iconManager.getTexture();
+
+      if (iconsTexture) {
+        this.state.model.setUniforms({outlineBuffer: 0}).draw();
```

Comment:
This uniform used to be `DEFAULT_BUFFER` when `outlineWidth: 0`. Is the change intentional?

Reply:
No, the intention is just to draw the text sans outline, I'll use `DEFAULT_BUFFER` instead